src/Dumplie/Application/Command/Customer/CreateCartHandler.php

```diff
@@ -38,11 +38,10 @@ public function __construct(Carts $carts, Factory $factory)
      */
     public function handle(CreateCart $command)
     {
-        $this->carts->add(new Cart(new CartId($command->uuid()), $command->currency()));
-
         $transaction = $this->factory->open();
 
         try {
+            $this->carts->add(new Cart(new CartId($command->uuid()), $command->currency()));
```

Comment:
Shouldn't a new UUID be generated here? I mean, it's meant to be Universally Unique, but here it's the same for command and for cart.